deployer

```diff
@@ -0,0 +1,220 @@
+        stages=stages,
+    )
+    return c
+
+
+def gen_badge(pkg, ver):
+    tpl = """
+<svg xmlns="http://www.w3.org/2000/svg" width="{width}" height="20">
+  <linearGradient id="b" x2="0" y2="100%">
+    <stop offset="0" stop-color="#bbb" stop-opacity=".1"/>
+    <stop offset="1" stop-opacity=".1"/>
+  </linearGradient>
+  <mask id="a">
+    <rect width="{width}" height="20" rx="3" fill="#fff"/>
+  </mask>
+  <g mask="url(#a)">
+    <path fill="#555" d="M0 0h{pkg_width}v20H0z"/>
+    <path fill="#97CA00" d="M{pkg_width} 0h{ver_width}v20H{pkg_width}z"/>
+    <path fill="url(#b)" d="M0 0h{width}v20H0z"/>
+  </g>
+  <g fill="#fff" font-family="DejaVu Sans,Verdana,Geneva,sans-serif" font-size="11">
+    <text x="4" y="15" fill="#010101" fill-opacity=".3">{pkg}</text>
+    <text x="4" y="14">{pkg}</text>
+    <text x="{ver_x}" y="15" fill="#010101" fill-opacity=".3">{ver}</text>
+    <text x="{ver_x}" y="14">{ver}</text>
+  </g>
+</svg>
+"""
+    pw = len(pkg) * 7 + 8
+    vw = len(ver) * 7 + 8
+    return tpl.format(
+        pkg=pkg, ver=ver, width=pw + vw, pkg_width=pw, ver_x=pw + 4, ver_width=vw
+    )
+
+
+def write_badge(path, hn, pkg, ver):
+    badge = gen_badge(pkg, ver)
+    with NamedTemporaryFile() as tmpf:
+        tmpf.write(badge.encode())
+        tmpf.flush()
+        cmd = ["scp", "-C", "-B", tmpf.name, f"{hn}:~/{pkg}.svg"]
+        subprocess.run(cmd, timeout=300)
+        cmd = ["ssh", hn, "sudo", "mv", f"{pkg}.svg", f"{path}/{pkg}.svg"]
+        subprocess.run(cmd, timeout=300)
+        cmd = ["ssh", hn, "sudo", "chmod", "a+r", f"{path}/{pkg}.svg"]
+        subprocess.run(cmd, timeout=300)
+
+
+def fetch_host_packages_versions(conf, hn):
+    o = {}
+    cmd = ["ssh", hn, "-T", "dpkg-query", "--show"] + conf["deb_packages"]
+    out = subprocess.run(cmd, timeout=30, stdout=PIPE)
+    for line in out.stdout.decode().splitlines():
+        if "\t" not in line:
+            continue
+        pkg, ver = line.split()
+        o[pkg] = ver
+
+    return o
+
+
+def fetch_packages_versions(conf):
+    status = {}  # stage -> hostname -> package -> version
+    for stage, hosts in conf["stages"].items():
+        status[stage] = {}
+        for hn in hosts:
+            status[stage][hn] = fetch_host_packages_versions(conf, hn)
+    return status
+
+
+def print_status(conf):
+    status = fetch_packages_versions(conf)
+
+    print()
+    stage_names = conf["stages"].keys()
+    hdr = "{:^15}  " * (len(stage_names) + 1)
+    print(hdr.format("Package", *stage_names))
+    # ➛ ➜ ➔ ➝ ➞ ➟ ➠ ➧ ➨  ►  ➢ ➣ ➤   ⟿     ✅
+    for pkg in conf["deb_packages"]:
+        tpl = "{:15}  "
+        versions = []
+        for stage in stage_names:
+            for hn in status[stage]:
+                ver = status[stage][hn].get(pkg, "")
+                if versions == []:
+                    tpl += "   {:15}"
+                else:
+                    prev = versions[-1]
+                    compare = apt_pkg.version_compare(prev, ver)
```

Comment:
Can we replace this with https://packaging.pypa.io/en/latest/version.html#packaging.version.parse?